blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -56,12 +117,69 @@ namespace blueprint
             _viewManager->performRootShadowTreeLayout();
         }
 
+        void paint(juce::Graphics& g) override
+        {
+            if (errorText)
+            {
+                g.fillAll(juce::Colour(0xffe14c37));
+                errorText->draw(g, getLocalBounds().toFloat().reduced(10.f));
+            }
+            else
+            {
+                View::paint(g);
+            }
+        }
+
         //==============================================================================
-        /** Evaluates a javascript bundle in the Ecmascript engine. */
+        /**
+         * Evaluates a javascript bundle/code in the Ecmascript engine.
+         *
+         * It is possible to pass a js bundle file to this evaluate() overload using juce::File::loadAsString. However,
+         * if you require hot-reload functionality you should use the juce::File based evaluate() overload.
+         * This overload can also be used to evaluate js code inline/directly.
+         **/
         juce::var evaluate (const juce::String& bundle)
         {
             JUCE_ASSERT_MESSAGE_THREAD
-            return engine.evaluate(bundle);
+
+            // Clear error state from previous js evals
+            errorText.reset();
+            auto result = engine.evaluate(bundle);
+
+            if (onBundleEvaluated)
+                onBundleEvaluated(std::nullopt);
```

Comment:
:+1:

Yea we definitiely want this method around still. In develeopment builds it's always preferable to just read the the bundle file from disk, which enables things like hot reloading and enables skipping a compile step to get webpack's latest output. But for shipping a product, what I've been doing is injecting the production bundle as a Binary Resource in the projucer, and reading it from memory as a string at runtime

Reply:
Right cool. My current use case installs the bundle on the target system along with our other application resources so I was missing this use-case. Cool. Will leave this as is